pallets/foreign-investments/src/swaps.rs

```diff
@@ -183,26 +155,25 @@ impl<T: Config> Swaps<T> {
 				} else {
 					let inverse_swap = swap;
 
-					let new_swap_amount_out = T::CurrencyConverter::stable_to_stable(
-						new_swap.currency_out,
+					let new_swap_amount_in = T::TokenSwaps::convert_by_market(
 						new_swap.currency_in,
-						new_swap.amount_in,
+						new_swap.currency_out,
+						new_swap.amount_out,
 					)?;
 
-					match inverse_swap.amount_in.cmp(&new_swap_amount_out) {
+					match inverse_swap.amount_out.cmp(&new_swap_amount_in) {
 						Ordering::Greater => {
 							let amount_to_swap =
-								inverse_swap.amount_in.ensure_sub(new_swap_amount_out)?;
+								inverse_swap.amount_out.ensure_sub(new_swap_amount_in)?;
 
 							T::TokenSwaps::update_order(
-								who.clone(),
 								swap_id,
 								amount_to_swap,
-								T::BalanceRatio::one(),
+								OrderRatio::Market,
 							)?;
 
 							Ok(SwapStatus {
-								swapped: new_swap.amount_in,
+								swapped: new_swap_amount_in,
```

Comment:
No, because `amount_to_swap` is the pending swap amount but the `status.swapped` returned by `apply_over_swap` refers to the amount that can be regarded as fulfilled swap because of resolving opposite swaps.

**Example**
Assume
* inverse swap is from pool to foreign with `amount_out = 100P`
* new swap is from foreign to pool with `amount_in = 30P`

Then, the decreasing swap from foreign to pool can be regarded as fulfilled such that the `status.swapped = 30P` and `amount_to_swap = 70P`

Reply:
I was referring to the line below, that was not changed. It is `zero` in this case, as I wrote `pending: amount_to_swap`

Reply:
No, because the `pending` here is seen from the perspective of whoever makes the swap. So, if you apply a swap, in this case, you do not have to wait for more "swapping" because your swap has been completely swaped. Independently, it could be any other inverse swap pending, but that doesn't concern you at this moment. You've got fully swapped your swap.

Reply:
I see. Thanks!